keps/sig-multicluster/2149-clusterid/README.md

```diff
@@ -0,0 +1,705 @@
+
+While a member of a ClusterSet, a cluster will also have an additional `clusterset.k8s.io ClusterClaim` which describes its current membership. This claim must be present exactly as long as the set membership lasts, and removed when the cluster is no longer a member.
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### ClusterSet membership
+
+I have some set of clusters working together and need a way to uniquely identify them within the system that I use to track membership.
+
+#### Multi-Cluster Services
+
+I have a headless multi-cluster service deployed across clusters in my ClusterSet with similarly named pods in each cluster. I need a way to disambiguate each backend pod via DNS.
+
+#### Diagnostics
+
+Clusters within my ClusterSet send logs/metrics to a common monitoring solution and I need to be able to identify the cluster from which a given set of events originated.
+
+
+#### Joining or moving between ClusterSets
+
+I want the ability to add a previously-isolated cluster to a ClusterSet, or to move a cluster from one ClusterSet to another and be aware of this change.
+
+### `ClusterClaim` CRD
+
+The `ClusterClaim` resource provides a way to store identification related, cluster scoped information for multi-cluster tools while creating flexibility for implementations. A cluster may have multiple `ClusterClaim`s, each holding a different identification related value. Each claim contains the following information:
+
+*   **Name** - a well known or custom name to identify the claim.
+*   **Value** - a claim-dependent string.
+
+
+### Well known claims
+
+The requirements below use the keywords **must, should,** and **may** purposefully in accordance with [RFC-2119](https://tools.ietf.org/html/rfc2119).
+
+
+#### Claim: `id.k8s.io`
+
+Contains a unique identifier for the containing cluster.
+
+
+##### Uniqueness
+
+*   The identifier **must** be unique within the ClusterSet to which its cluster belongs for the duration of the cluster’s membership.
+*   The identifier **should** be unique beyond the ClusterSet within the scope of expected use.
+*   The identifier **may** be globally unique beyond the scope of its ClusterSet.
+*   The identifier **may** be unique beyond the span of its cluster’s membership and lifetime.
+
+
+##### Lifespan
+
+*   The identifier **must** exist and be immutable for the duration of a cluster’s membership in a ClusterSet, and as long as a `clusterset.k8s.io` claim exists.
+*   The identifier **must** exist for the lifespan of a cluster.
+*   The identifier **may** be immutable for the lifespan of a cluster.
+
+
+##### Contents
+
+*   The identifier **must** be a valid RFC-1123 DNS label.
```

Comment:
Personally I suspect many folks will want to use a subdomain here instead of just a label. I would be okay with starting with labels and expanding to use subdomains later if it makes things easier.

Reply:
Expanding like that is a breaking change - the value might be used in contexts where a subdomain is not valid.  If we think we have use-cases, we should put some thought into this sooner than later.

Reply:
Added to alpha->beta graduation criteria in 793c08b